test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -0,0 +1,856 @@
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						url: {
+							source: 'core/post-meta',
+							args: { key: 'url_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.altOnly = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						alt: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.titleOnly = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						title: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+		variables.blocks.images.multipleAttrs = {
+			name: 'core/image',
+			attributes: {
+				url: variables.placeholderSrc,
+				alt: 'default alt value',
+				title: 'default title value',
+				metadata: {
+					bindings: {
+						url: {
+							source: 'core/post-meta',
+							args: { key: 'url_custom_field' },
+						},
+						alt: {
+							source: 'core/post-meta',
+							args: { key: 'text_custom_field' },
+						},
+					},
+				},
+			},
+		};
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllMedia();
+		await requestUtils.activateTheme( 'twentytwentyone' );
+		await requestUtils.deactivatePlugin( 'gutenberg-test-block-bindings' );
+	} );
+
+	test.describe( 'Template context', () => {
+		test.beforeEach( async ( { admin, editor } ) => {
+			await admin.visitSiteEditor( {
+				postId: 'emptytheme//index',
+				postType: 'wp_template',
+			} );
+			await editor.canvas.locator( 'body' ).click();
+			await editor.openDocumentSettingsSidebar();
+		} );
+
+		test.describe( 'Paragraph', () => {
+			test( 'Should show the value of the custom field', async ( {
+				editor,
+			} ) => {
+				await editor.insertBlock( variables.blocks.paragraph );
+				const paragraphBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Paragraph',
+				} );
+				const paragraphContent = await paragraphBlock.textContent();
+				expect( paragraphContent ).toBe(
+					variables.customFields.textKey
+				);
+			} );
+
+			test( 'Should lock the appropriate controls', async ( {
+				editor,
+				page,
+			} ) => {
+				await editor.insertBlock( variables.blocks.paragraph );
+				const paragraphBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Paragraph',
+				} );
+				await paragraphBlock.click();
+
+				// Alignment controls exist.
+				await expect(
+					page.getByRole( 'button', {
+						name: variables.labels.align,
```

Comment:
Nit: I normally prefer inlining these variables and primitives to make the tests more readable. Also, we can nest locators to make it clearer that it's a control in the block settings.

Reply:
I've removed the variables in this commit as part of a follow-up pull request: [link](https://github.com/WordPress/gutenberg/commit/7091e845da675c81606a5fae36280a7a99367481).

Additionally, I nested the locators in this other one: [link](https://github.com/WordPress/gutenberg/commit/1c0f76819ce9ca6302d9321e4ddd81d19353a33f).